tools/cldr-apps/src/main/java/org/unicode/cldr/web/XPathTable.java

```diff
@@ -758,24 +756,7 @@ public String getByStringID(String id) {
         }
     }
 
-    String getByStringID(long l) {
-        String s = sidToString.get(l);
-        if (s != null) return s;
-        // slow way
-        for (String x : stringToId.keySet()) {
-            if (getStringID(x) == l) {
-                sidToString.put(l, x);
-                return x;
-            }
-        }
-        if (SurveyMain.isUnofficial()) {
-            logger.warning(
-                    "xpt: Couldn't find stringid "
-                            + Long.toHexString(l)
-                            + " - sid has "
-                            + sidToString.size());
-        }
-        // it may be
-        return null;
+    final String getByStringID(long l) {
+        return StringId.getStringFromId(l);
```

Comment:
this is the core problem.  I don't know why XPathTable had its own StringId cache.